torchvision/transforms/functional_tensor.py

```diff
@@ -636,3 +620,110 @@ def affine(
         img = torch.round(img).to(out_dtype)
 
     return img
+
+
+def affine(
+        img: Tensor, matrix: List[float], resample: int = 0, fillcolor: Optional[int] = None
+) -> Tensor:
+    """Apply affine transformation on the Tensor image keeping image center invariant.
+
+    Args:
+        img (Tensor): image to be rotated.
+        matrix (list of floats): list of 6 float values representing inverse matrix for affine transformation.
+        resample (int, optional): An optional resampling filter. Default is nearest (=0). Other supported values:
+            bilinear(=2).
+        fillcolor (int, optional): this option is not supported for Tensor input. Fill value for the area outside the
+            transform in the output image is always 0.
+
+    Returns:
+        Tensor: Transformed image.
+    """
+    _interpolation_modes = {
+        0: "nearest",
+        2: "bilinear",
+    }
+
+    _assert_grid_transform_inputs(img, matrix, resample, fillcolor, _interpolation_modes)
+
+    theta = torch.tensor(matrix, dtype=torch.float).reshape(1, 2, 3)
+    shape = img.shape
+    grid = affine_grid(theta, size=(1, shape[-3], shape[-2], shape[-1]), align_corners=False)
+    mode = _interpolation_modes[resample]
+
+    return _apply_grid_transform(img, grid, mode)
+
+
+def _compute_output_size(theta: Tensor, w: int, h: int) -> Tuple[int, int]:
```

Comment:
Do we need this (relatively expensive) function for calculating the output size? I think we can get it in constant time.

From some quick analysis, we might be able to get the height/width of the new image via something like
```python
h_new = h * sin(alpha) + w * cos(alpha)
w_new = h * cos(alpha) + w * sin(alpha)
```
(but math should be verified, and angles might be off)
![20200729_112911](https://user-images.githubusercontent.com/9110200/88783473-1c16b480-d18f-11ea-9b43-91ad31ff2379.jpg)

Reply:
Analytical approach can be nice :) 
Here there matrix computations for just 4 points, so it is not that expensive. Let me check if I can provide an analytical computation with the same result as PIL.